### PR TITLE
Register

### PR DESCRIPTION
**Pull request by bagdaddy**

### CODE REVIEW

**Comment by domanttas** [comment]
@tomasmuzas help me with this PR :)

**Comment by domanttas** [comment]
@bagdaddy apparently there are few merge conflicts, they need to be resolved

File: WhoIsThat/WhoIsThat/ViewModels/LoginViewModel.cs [hunk 1]
```diff
@@ -1,25 +1,115 @@
...
+        { 
+            //Checks if all required fields are filled
+
+            /*if(PersonObject.PersonFirstName == null || PersonObject.PersonLastName == null || PersonObject.DescriptiveSentence == null || takenPhoto == null)
+            {
+                return;
+            }*/
+
+            //If so, stores his information in the db
+            PersonObject.Id = 1;
```
Comment (domanttas) [inline WhoIsThat/WhoIsThat/ViewModels/LoginViewModel.cs]: ID is automatically generated in backend, so you don't need to specify it here

File: WhoIsThat/WhoIsThat/ViewModels/LoginViewModel.cs [hunk 2]
```diff
@@ -1,25 +1,115 @@
...
+            //Save taken photo to Azure cloud for recognition, later on it is deleted
+
+            //await CloudStorageService.SaveBlockBlob(takenPhoto);
+        }
+
+        public async void SavePerson()
+        { 
+            //Checks if all required fields are filled
+
+            /*if(PersonObject.PersonFirstName == null || PersonObject.PersonLastName == null || PersonObject.DescriptiveSentence == null || takenPhoto == null)
```
Comment (domanttas) [inline WhoIsThat/WhoIsThat/ViewModels/LoginViewModel.cs]: So there's no handling for scenario if user left the fields blank?

File: WhoIsThat/WhoIsThat/ViewModels/LoginViewModel.cs [hunk 3]
```diff
@@ -1,25 +1,115 @@
...
+            }
+            Application.Current.Properties.Add("UserRegistered", true);
+            await Application.Current.SavePropertiesAsync();
         }
 
         public async void NavigateToHomePage()
         {
             await Application.Current.MainPage.Navigation.PushAsync(new HomePage(new HomeViewModel()));
         }
+        public void OnPropertyChanged([CallerMemberName] string propertiesName = "")
```
Comment (domanttas) [inline WhoIsThat/WhoIsThat/ViewModels/LoginViewModel.cs]: Is this really better than it was before?

@tomasmuzas maybe you could check this feature out

File: WhoIsThat/WhoIsThat/ViewModels/LoginViewModel.cs (hunk 1 above)
Comment (bagdaddy) [reply WhoIsThat/WhoIsThat/ViewModels/LoginViewModel.cs]: true, i used this to see, if the app remembers the id stored in application properties. It's removed now

File: WhoIsThat/WhoIsThat/ViewModels/LoginViewModel.cs (hunk 3 above)
Comment (bagdaddy) [reply WhoIsThat/WhoIsThat/ViewModels/LoginViewModel.cs]: screenshot_1
There are a lot of ways to implement OnPropertyChanged method and they don't really differ much.  This implementation accepts arguments (calling OnPropertyChanged("someProperty") is acceptable), but can also take the callers name as an argument. Similar would be OnPropertyChanged(nameof(value)) in this case. 

However, i'm not even sure this feature is required here, even though all sources suggest it is. I tried filling the fields with and without triggering this method, and it turns out fine either way.

**Pull request merged**